tests/Integ/S3Context.php

```diff
@@ -6,19 +6,195 @@
+                ["bucket" => self::getResourceName()],
+                ["starts-with", '$key', ""],
+                ["acl" => "public-read"],
+                ["x-amz-credential" => $credentails],
+                ["x-amz-algorithm" => "AWS4-HMAC-SHA256"],
+                ["x-amz-date" => $ldt],
+            ],
+        ];
+        $postObjectV4 = new PostObjectV4(
+            $this->s3Client,
+            self::getResourceName(),
+            $policyV4['conditions']
+        );
+        $this->preparePostData($postObjectV4, 'V4');
+    }
+
+    /**
+     * @When I make a HTTP POST request
+     */
+    public function iMakeAHttpPostRequest()
+    {
+        //V2
+        try {
+            (new Client)->request(
+                $this->attributes['V2']['method'],
+                $this->attributes['V2']['action'],
+                [
+                    'multipart' => $this->inputs['V2'],
+                ]
+            );
+        } catch (\GuzzleHttp\Exception\ClientException $e) {
+            echo $e->getResponse()->getBody();
+        }
+
+        // V4
+        try {
+            (new Client)->request(
+                $this->attributes['V4']['method'],
+                $this->attributes['V4']['action'],
+                [
+                    'multipart' => $this->inputs['V4'],
+                ]
+            );
+        } catch (\GuzzleHttp\Exception\ClientException $e) {
+            echo $e->getResponse()->getBody();
+        }
+    }
+
+    /**
+     * @Then a file is uploaded
+     */
+    public function aFileIsUploaded()
+    {
+        Assert::assertTrue($this->s3Client->doesObjectExist(
+            self::getResourceName(),
+            'fileV2.ext'
+        ));
+        Assert::assertTrue($this->s3Client->doesObjectExist(
+            self::getResourceName(),
+            'fileV4.ext'
+        ));
+    }
+
+    /**
+     * Prepare form inputs and attribute for POST
+     */
+    private function preparePostData($postObject, $version)
+    {
+        $this->attributes[$version] = $postObject->getFormAttributes();
+        foreach ($postObject->getFormInputs() as $name => $contents) {
+            $this->inputs[$version] []= [
+                'name' => $name,
+                'contents' => $contents,
+            ];
+        }
+        $this->inputs[$version] []= [
+            'name' => 'file',
+            'contents' => $this->stream,
+            'filename' => 'file'. $version .'.ext',
+        ];
+    }
+
+    private static function getResourceName()
+    {
+        static $bucketName;
+
+        if (empty($bucketName)) {
+            $bucketName = self::getResourcePrefix() . self::RESOURCE_POSTFIX;
+        }
+
+        return $bucketName;
+    }
+
+    /**
+     * @BeforeSuite
+     */
+    public static function createTempFile()
+    {
+        self::$tempFile = tempnam(sys_get_temp_dir(), self::getResourceName());
+        file_put_contents(self::$tempFile, str_repeat('x', 10 * self::MB + 1024));
```

Comment:
Is there a reason the file is so large? This could take a couple of seconds to upload when running the tests on spotty wifi. It's not a huge deal, but I think a ~128K file exercises the same functionality. Also, there's no real need for a constant here: just `128 * 1024` is fine.